library/src/main/java/com/getstream/sdk/chat/ChatImpl.java

```diff
@@ -109,8 +100,23 @@ public String getVersion() {
         return BuildConfig.BUILD_TYPE + ":" + BuildConfig.VERSION_NAME;
     }
 
-    void init() {
+    @Override
+    public void setUser(@NotNull User user, @NotNull String token, @NotNull InitConnectionListener callbacks) {
+        ChatClient.instance().setUser(user, token, new InitConnectionListener() {
```

Comment:
If we remove `ChatClient` as an attribute and delegate it on the singleton, we are assuming the user know that it needs to be initialized by himself. But it is a bit weird that I need to init another different class to be able to use `ChatDomain`'s methods. 
Why don't we ask for it on the `ChatDomain.Builder` or have any way to init it internally?

Reply:
Imo, there's no reason for saving references to `ChatClient` and `ChatDomain` instances into any class fields (like before), as they are public, global singleton instances. For me, they are designed in a way that allows users to get them anywhere.

I think the one root problem here with the `Chat.setUser()` function is related to `ChatDomain`'s design. `ChatDomain` stores a duplicate `User` property. This should be redundant since it also uses `ChatClient` which has the user set.  So at the very beginning, `User` should be set only in one place and, imo, it should be the `ChatClient`.

Reply:
@tschellenbach need your opinion here whether it would be possible to instantiate `ChatDomain` without passing a `User` to it ☝️

Reply:
Yep, `Chat` shouldn't touch client at all. Layers must not be mixed.
`Chat` is a dumb ui. It should just consume data from domain and pass user events.
```kotlin
Chat {
  fun setUser(user:User) {
    domain.setUser(user)
  }
}

ChatDomain {
  fun setUser(user: User){
    // sync or replace
   client.setUser(user)
  }
}
```

Reply:
Looks good to me. We'd need to provide `ChatDomain.setUser()` then. 

cc @tschellenbach